src/psyclone/parse/algorithm.py

```diff
@@ -0,0 +1,976 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2019, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford and A. R. Porter STFC Daresbury Lab
+
+'''Module that uses the Fortran parser fparser2 to parse
+PSyclone-conformant Algorithm code.
+
+'''
+
+from __future__ import absolute_import
+
+from psyclone.configuration import Config
+from psyclone.parse.utils import check_api, check_ll, ParseError
+from psyclone.psyGen import InternalError
+
+from fparser.common.readfortran import FortranFileReader
+from fparser.two import pattern_tools
+from fparser.two.utils import walk_ast, FortranSyntaxError
+from fparser.two.parser import ParserFactory
+# pylint: disable=no-name-in-module
+from fparser.two.Fortran2003 import Main_Program, Module, \
+    Subroutine_Subprogram, Function_Subprogram, Use_Stmt, \
+    Call_Stmt, Actual_Arg_Spec_List, Actual_Arg_Spec, Part_Ref, \
+    Only_List, Char_Literal_Constant, Section_Subscript_List, \
+    Name, Real_Literal_Constant, Data_Ref, Int_Literal_Constant, \
+    Function_Reference
+# pylint: enable=no-name-in-module
+
+#
```

Comment:
There are 3 related comments in this file which indicate different sections. This comment is the first of the three. I've added `Section x:` to each comment for clarity, rather than removing the comments.

Reply:
OK - thanks.